numpy/_core/src/multiarray/convert_datatype.cpp

```diff
@@ -2121,13 +2115,13 @@ simple_cast_resolve_descriptors(
 
     loop_descrs[0] = NPY_DT_CALL_ensure_canonical(given_descrs[0]);
     if (loop_descrs[0] == NULL) {
-        return -1;
+        return (NPY_CASTING)-1;
```

Comment:
Probably for another PR, but I wonder if one shouldn't just do `return _NPY_ERROR_OCCURRED_IN_CAST;`?